### PR TITLE
chore: [IOBP-142] Removal of NativeBase in InitiativeSettings

### PR DESCRIPTION
**Pull request by forrest57**
## Short descriptions
removed NativeBase from InitiativeSettings component
Screenshot 2023-07-21 at 11 32 31

## List of changes proposed in this pull request
- moved from custom nativebase component to DS ListItems for the InitiativeSettings component (add card/add Iban ListItems)
- fixed ListItemNav typo that broke the component if passed a ReactElement

## How to test
navigate to a refund-type initiative (non [D] when using dev-server) and make sure both List Items render correctly both in error and not in error, and that when pressed the animation extends to the whole screen's width.
also, check both with and without the experimental DS

### CODE REVIEW

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 1]
```diff
@@ -131,11 +76,17 @@ const InitiativeSettingsComponent = (props: Props) => {
...
-          hasWarnings={status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_IBAN}
-        />
-      )
+          ),
+
+          onPress: () => navigateToInstrumentsConfiguration(initiativeId),
+          accessibilityLabel: ""
+        };
+        return status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_IBAN ? (
+          <ListItemNavAlert {...listItemOptions} />
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:85]:
```suggestion
          <ListItemNavAlert
            {...listItemOptions}
            description={I18n.t(
              "idpay.initiative.details.initiativeDetailsScreen.configured.settings.actionsRequired"
            )}
          />
```

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 2]
```diff
@@ -144,25 +95,31 @@ const InitiativeSettingsComponent = (props: Props) => {
...
-        />
-      )
+          ),
+          description: iban,
+          onPress: () => navigateToIbanConfiguration(initiativeId),
+          accessibilityLabel: ""
+        };
+        return status ===
+          InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_INSTRUMENT ? (
+          <ListItemNavAlert {...listItemOptions} />
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:118]:
```suggestion
          <ListItemNavAlert
            {...listItemOptions}
            description={I18n.t(
              "idpay.initiative.details.initiativeDetailsScreen.configured.settings.actionsRequired"
            )}
          />
```

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 3]
```diff
@@ -131,11 +76,17 @@ const InitiativeSettingsComponent = (props: Props) => {
...
+          ),
+
+          onPress: () => navigateToInstrumentsConfiguration(initiativeId),
+          accessibilityLabel: ""
+        };
+        return status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_IBAN ? (
+          <ListItemNavAlert {...listItemOptions} />
+        ) : (
+          <ListItemNav {...listItemOptions} />
+        );
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:88]:
```suggestion
        const areActionsRequired =
          status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_IBAN ||
          status === InitiativeStatusEnum.NOT_REFUNDABLE;

        return areActionsRequired ? (
          <ListItemNavAlert
            {...listItemOptions}
            description={I18n.t(
              "idpay.initiative.details.initiativeDetailsScreen.configured.settings.actionsRequired"
            )}
          />
        ) : (
          <ListItemNav {...listItemOptions} />
        );
```

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 4]
```diff
@@ -144,25 +95,31 @@ const InitiativeSettingsComponent = (props: Props) => {
...
+          description: iban,
+          onPress: () => navigateToIbanConfiguration(initiativeId),
+          accessibilityLabel: ""
+        };
+        return status ===
+          InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_INSTRUMENT ? (
+          <ListItemNavAlert {...listItemOptions} />
+        ) : (
+          <ListItemNav {...listItemOptions} />
+        );
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:121]:
```suggestion
        const areActionsRequired =
          status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_INSTRUMENT ||
          status === InitiativeStatusEnum.NOT_REFUNDABLE;

        return areActionsRequired ? (
          <ListItemNavAlert
            {...listItemOptions}
            description={I18n.t(
              "idpay.initiative.details.initiativeDetailsScreen.configured.settings.actionsRequired"
            )}
          />
        ) : (
          <ListItemNav {...listItemOptions} />
        );
```

**Review by mastro993 — CHANGES_REQUESTED** [review]
The "Actions required" label is gone missing after this change.
Also, the label should be displayed in case of `NOT_REFUNDABLE` initiative.

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 5]
```diff
@@ -131,11 +76,17 @@ const InitiativeSettingsComponent = (props: Props) => {
...
             }
-          )}
-          onPress={() => navigateToInstrumentsConfiguration(initiativeId)}
-          hasWarnings={status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_IBAN}
-        />
-      )
+          ),
+
+          onPress: () => navigateToInstrumentsConfiguration(initiativeId),
+          accessibilityLabel: ""
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:82]: We should add the `accessibilityLabel` prop for a11y

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx [hunk 6]
```diff
@@ -144,25 +95,31 @@ const InitiativeSettingsComponent = (props: Props) => {
...
-          onPress={() => navigateToIbanConfiguration(initiativeId)}
-          hasWarnings={
-            status === InitiativeStatusEnum.NOT_REFUNDABLE_ONLY_INSTRUMENT
-          }
-        />
-      )
+          ),
+          description: iban,
+          onPress: () => navigateToIbanConfiguration(initiativeId),
+          accessibilityLabel: ""
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:114]: Same as above

**Review by mastro993 — CHANGES_REQUESTED** [review]

**Comment by forrest57** [comment]
@mastro993 's comments were addressed in [7dd5eb7](https://github.com/pagopa/io-app/pull/4827/commits/7dd5eb7055bff45a170137af77c4da2588994413)

File: ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx (hunk 5 above)
Comment (mastro993) [reply ts/features/idpay/initiative/details/components/InitiativeSettingsComponent.tsx:82]: @forrest57 still missing after latest commit!

**Review by mastro993 — CHANGES_REQUESTED** [review]

**Pull request merged**